optuna/samplers/_cmaes.py

```diff
@@ -288,6 +302,25 @@ def _log_independent_sampling(self, trial: FrozenTrial, param_name: str) -> None
             )
         )
 
+    def _get_trials(self, study: "optuna.Study") -> List[FrozenTrial]:
```

Comment:
How about calling `warnings.warn("...", ExperimentalWarning)` if `self._consider_pruned_trials` is True.